mlflow/pyfunc/__init__.py

```diff
@@ -828,6 +828,10 @@ def _create_model_downloading_tmp_dir(should_use_nfs):
 _MLFLOW_SERVER_OUTPUT_TAIL_LINES_TO_KEEP = 200
 
 
+def _parse_spark_datatype(spark, datatype: str):
+    return spark.createDataFrame([], datatype).schema[0].dataType
```

Comment:
Hold on.

Seemingly there are some problems.

Suppose user specify  UDF result type as `a long, b int`,
then `spark.createDataFrame([], 'a long, b int').schema[0].dataType` returns `LongType()` but we expect it returns a struct type

Reply:
another way is `udf(lambda x: x, returnType=datatype).returnType`. In that way, it will return the type as is.

Reply:
lets use udf

Reply:
FYI: I am adding `DataType.fromDDL` that can replace this (https://github.com/apache/spark/pull/43226)